### PR TITLE
Add complex autograd support and OpInfo based test for torch.addr

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50667 Add complex autograd support and OpInfo based test for torch.addr**

Differential Revision: [D25957584](https://our.internmc.facebook.com/intern/diff/D25957584)

### CODE REVIEW

**Comment by mruberry** [comment]
Two builds are failing because of test_variant_consistency_jit_addr_cuda_bfloat16 with

```
RuntimeError: at::cuda::blas::gemv: not implemented for N3c108BFloat16E
```

@zasdfgbnm @ngimel we have a sip to recommend for this, right? Maybe we should think about giving that skip a clearer name so it's easier to read and discover.

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -810,6 +840,13 @@ def sample_inputs_fliplr_flipud(op_info, device, dtype, requires_grad):
                SkipInfo('TestCommon', 'test_variant_consistency_jit',
                         dtypes=[torch.bfloat16, torch.float16, torch.cfloat, torch.cdouble]),),
            sample_inputs_func=sample_inputs_addmm),
+    OpInfo('addr',
+           dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16, torch.float16),
+           skips=(
+               SkipInfo('TestCommon', 'test_variant_consistency_jit',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:846]: bfloat16 CUDA skip just needs to be added

**Review by mruberry — APPROVED** [review]
Just need to skip the one failing test.

Edit: oops, one other thing. The existing method_test entries should be incorporated into addr's sample inputs and removed from method_tests:

https://github.com/pytorch/pytorch/blob/7e05d07ca7557b0b433b97a5943c8eedbbf34837/torch/testing/_internal/common_methods_invocations.py#L2065

**Comment by anjali411** [comment]
> Just need to skip the one failing test.
> 
> Edit: oops, one other thing. The existing method_test entries should be incorporated into addr's sample inputs and removed from method_tests:
> 
> https://github.com/pytorch/pytorch/blob/7e05d07ca7557b0b433b97a5943c8eedbbf34837/torch/testing/_internal/common_methods_invocations.py#L2065

are you suggesting to add the sample inputs to test broadcasting in the `torch.addr` `OpInfo` based test and skip `est_inplace_grad` and `test_variant_consistency_eager` for `all dtypes`?

I didn't delete the `method_test` entry because I didn't add broadcasting tests in the new `OpInfo` based test.

**Comment by mruberry** [comment]
> are you suggesting to add the sample inputs to test broadcasting in the `torch.addr` `OpInfo` based test and skip `test_inplace_grad` and `test_variant_consistency_eager` for `all dtypes`?

Yes.

> I didn't delete the `method_test` entry because I didn't add broadcasting tests in the new `OpInfo` based test. I also saw that `torch.addmm` method_tests haven't been removed yet even though a new `OpInfo` based test (which does not test broadcasting, as tested in `method_tests`) has been added for it.

That makes sense, but I think it's OK to move and skip in this case. We don't need to update addmm here.

**Pull request closed without merging**